samples/matter/light_bulb/src/app_task.cpp

```diff
@@ -196,6 +190,20 @@ CHIP_ERROR AppTask::Init()
 	}
 	mPWMDevice.SetCallbacks(ActionInitiated, ActionCompleted);
 
+#ifdef CONFIG_CHIP_OTA_REQUESTOR
+	/* OTA image confirmation must be done before the factory data init. */
+	err = OtaConfirmNewImage();
+	if (err != CHIP_NO_ERROR) {
+		return err;
+	}
+#endif
+
+#ifdef CONFIG_MCUMGR_TRANSPORT_BT
+	/* Initialize DFU over SMP */
+	GetDFUOverSMP().Init();
+	GetDFUOverSMP().ConfirmNewImage();
```

Comment:
Is it sane to double confirm the image in case we have both types of DFU supported in the application?

Reply:
Also, we do the same in OTAImageProcessorImpl::ConfirmCurrentImage(), so the generic question would be if it's all right to call `boot_write_img_confirmed()` multiple times at boot up. I guess it may simply fail, but maybe the propagated error should be handled somehow differently?

Reply:
Regarding documentation of `boot_write_image_confirmed` function:
```
 * This routine is safe to call if the current image has already been
 * confirmed. It will return a successful result in this case.
```
So we can call this several times.
And regarding merging it with BLE DFU, apart from calling `boot_write_image_confirmed` it also sets a flag in the OTA image processor so it can not be called together with BLE DFU confirmation.